File: src/patches/clawpatch/fort_user/fclaw2d_clawpatch_exceeds_threshold.c
/* # check to see if value exceeds threshold */

#include <fclaw2d_clawpatch.h>

#include <fclaw2d_clawpatch_options.h>
#include <fclaw2d_clawpatch_fort.h>

int FCLAW2D_CLAWPATCH_EXCEEDS_THRESHOLD(int* blockno,
                                        double qval[], 
                                        double* qmin, double *qmax,
                                        double quad[], 
                                        double *dx, double *dy, 
                                        double *xc, double *yc, 
                                        int* tag_threshold,
                                        int* init_flag,
                                        int* is_ghost)
{
#if 0

    int refinement_criteria = fclaw2d_clawpatch_get_refinement_criteria();

    int exceeds_th;

    fclaw2d_clawpatch_vtable_t* clawpatch_vt = fclaw2d_clawpatch_vt();
    clawpatch_fort_exceeds_threshold_t user_exceeds_threshold = 
                                clawpatch_vt->fort_user_exceeds_threshold;

    exceeds_th = 1;
    switch(refinement_criteria)
    {
        case FCLAW_REFINE_CRITERIA_VALUE:
            exceeds_th = FCLAW2D_CLAWPATCH_VALUE_EXCEEDS_TH(blockno,
                    qval,qmin,qmax,quad, dx, dy, xc, yc, 
                    tag_threshold,init_flag,is_ghost);
            break;
        case FCLAW_REFINE_CRITERIA_DIFFERENCE:
            exceeds_th = FCLAW2D_CLAWPATCH_DIFFERENCE_EXCEEDS_TH(blockno,
                    qval,qmin,qmax,quad, dx, dy, xc, yc, 
                    tag_threshold,init_flag,is_ghost);
            break;
        case FCLAW_REFINE_CRITERIA_MINMAX:
            exceeds_th = FCLAW2D_CLAWPATCH_MINMAX_EXCEEDS_TH(blockno,
                    qval,qmin,qmax,quad, dx, dy, xc, yc, 
                    tag_threshold,init_flag,is_ghost);

            break;
        case FCLAW_REFINE_CRITERIA_GRADIENT:
            exceeds_th = FCLAW2D_CLAWPATCH_GRADIENT_EXCEEDS_TH(blockno,
                    qval,qmin,qmax,quad, dx, dy, xc, yc, 
                    tag_threshold,init_flag,is_ghost);
            break;

        case FCLAW_REFINE_CRITERIA_USER:

            /* This must be define by the user, or nothing happens */
            if (user_exceeds_threshold == NULL)
            {
                fclaw_global_essentialf("\nfclaw2d_clawpatch_exceeds_threshold : " \
                                        "User specified criteria is set, but no user " \
                                        "defined\n function was found.\n\n");
            }
            FCLAW_ASSERT(user_exceeds_threshold != NULL);
            exceeds_th = user_exceeds_threshold(blockno, qval,qmin,qmax,quad, 
                                                dx, dy, xc, yc,
                                                tag_threshold,init_flag,is_ghost);
            break;
        default:
            fclaw_global_essentialf("fclaw2d_clawpatch_exceeds_threshold.c): " \
                                    "No valid refinement criteria specified\n");
            exit(0);
            break;
    }
    return exceeds_th;
#endif
}

File: src/patches/clawpatch/fort_user/fclaw2d_clawpatch_exceeds_threshold.f90
!! # This file isn't currently used;  The C version is used
!! # instead.

logical function fclaw2d_clawpatch_exceeds_threshold(blockno,& 
                                     qval,qmin,qmax,quad, & 
                                     dx,dy,xc,yc,tag_threshold,
                                     is_ghost)
    implicit none
    
    double precision :: qval,qmin,qmax,tag_threshold
    double precision :: quad(-1:1,-1:1)
    double precision :: dx,dy, xc, yc
    integer :: blockno, is_ghost
    logical :: refine

    integer :: refine_criteria
    common /com_refine/ refine_criteria

    logical :: exceeds_th
    logical :: fclaw2d_clawpatch_gradient_exceeds_th
    logical :: fclaw2d_clawpatch_difference_exceeds_th
    logical :: fclaw2d_clawpatch_value_exceeds_th
    logical :: fclaw2d_clawpatch_minmax_exceeds_th

    if (refine_criteria .eq. 0) then 
        exceeds_th = fclaw2d_clawpatch_value_exceeds_th(blockno, & 
                    qval,qmin,qmax,quad, dx,dy,xc,yc, & 
                    tag_threshold,is_ghost)
    else if (refine_criteria .eq. 1) then
        exceeds_th = fclaw2d_clawpatch_minmax_exceeds_th(blockno, & 
                    qval,qmin,qmax,quad, dx,dy,xc,yc, & 
                    tag_threshold,is_ghost)
    else if (refine_criteria .eq. 2) then
        exceeds_th = fclaw2d_clawpatch_difference_exceeds_th(blockno, &
                    qval,qmin,qmax,quad, dx,dy,xc,yc, & 
                    tag_threshold,is_ghost)
    else if (refine_criteria .eq. 3) then
        exceeds_th = fclaw2d_clawpatch_gradient_exceeds_th(blockno, & 
                    qval,qmin,qmax,quad, dx,dy,xc,yc, & 
                    tag_threshold,is_ghost)
    end if

    transport_exceeds_threshold = exceeds_th

end function transport_exceeds_threshold
